.gitignore

```diff
@@ -1,100 +1,94 @@
 # macOS
 **/*.DS_Store
 
-# Visual Studio Code
+# IDE & Editor Configuration
+## Visual Studio Code
 .vscode/*
 !.vscode/*.example
 .devcontainer/
 
-# JetBrains
+## JetBrains IDEs
 .idea/
 *.iml
 .fleet/
 
-# GitPod
+## Other Editors & Tools
 .gitpod.yml
-
-# CLion
 cmake-build-debug/
 /cmake-build-wsl_profile/
 
-# build
-*.exe
-*.app
-build/
-
-# Python
+# Programming Language Artifacts
+## Python
 *.pyc
 venv/
 
-# Golang
+## Go
 go/bin/
 
-# Rust
+## Rust - Compilation Artifacts
 src/**/target
 src/**/target_tarpaulin
 target/
 target-bisector-*/
+
+## Rust - Generated Code (Protocol Buffers)
 src/proto/
 src/prost/src/*.rs
 src/prost/src/sim/*.rs
 !src/prost/src/lib.rs
 !src/prost/src/imp.rs
 **/file_descriptor_set.bin
-# runtime log
-src/log/
-# srcy-tags
-**/srcy-tags.vi
 
-# rustc ice
+## Rust - Build Outputs & Binaries
```

Comment:
I don't think these are general items to be ignored. All build output are in the `target` folder by default. 

From the PR description, the motivation to ignore these items is that the pre-built risingwave binaries are downloaded to the root folder of source repo, which I think is not a common practice. We expect that users either download the pre-built binaries to somewhere clean to run it directly, or download the source and build the binaries from source.